lib/ansible/plugins/become/machinectl.py

```diff
@@ -81,7 +81,7 @@ def build_become_command(self, cmd, shell):
         if not cmd:
             return cmd
 
-        become = self._get_option('become_exe') or self.name
-        flags = self.get_option('flags') or ''
+        become = self.get_option('become_exe') or self.name
+        flags = self.get_option('become_flags') or ''
         user = self.get_option('become_user') or ''
-        return '%s shell -q %s %s@ -- %s' % (become, flags, user, cmd)
+        return '%s shell -q %s %s@ %s' % (become, flags, user, cmd)
```

Comment:
This change was explicitly done in https://github.com/ansible/ansible/pull/56572

Is this change really necessary? Also, maybe it's depending on the machinectl version? @ttencate any opinion on this?

Reply:
When running with systemd v232 (latest on Debian stretch) I get the following error when running with the `--` in the command:
```
fatal: [vps]: FAILED! => {
    "ansible_facts": {},
    "changed": false,
    "msg": "The following modules failed to execute: setup\n  setup: MODULE FAILURE\nSee stdout/stderr for the exact error\n"
}
```

and verbose logging displays the error as:
```
(1, b"Failed to get shell PTY: Specified path '--' is not absolute\r\n", b'Shared connection to 163.172.132.225 closed.\r\n')
```

When executing directly on the server as well I get the same issue: ```
joseph@cactus:~$ sudo machinectl shell -q -M buildContainer root@ /bin/echo Hello
Hello
joseph@cactus:~$ sudo machinectl shell -q -M buildContainer root@ -- /bin/echo Hello
Failed to get shell PTY: Specified path '--' is not absolute
```

Reply:
I did some more checking on this, and it seems that my fix was wrong: https://github.com/ansible/ansible/issues/56571#issuecomment-508671679

I agree that this should be reverted for now. Could someone also reopen #56571 in that case?

Reply:
Moved the `-q` argument to prepend the shell argument in the latest commit.

Reply:
Thanks! Note that that only fixes it for v230, not for versions < v230, so #56571 should still be reopened.